src/chainrules.jl

```diff
@@ -1,121 +1,152 @@
+    dims_args = (dims === nothing) ? () : (dims,)
+    true_dims = (dims === nothing) ? (1:ndims(x)) : dims 
+    y = rfft(x, dims_args...)
 
     # compute scaling factors
-    halfdim = first(dims)
+    halfdim = first(true_dims)
     d = size(x, halfdim)
     n = size(y, halfdim)
     scale = reshape(
         [i == 1 || (i == n && 2 * (i - 1) == d) ? 1 : 2 for i in 1:n],
-        ntuple(i -> i == first(dims) ? n : 1, Val(ndims(x))),
+        ntuple(i -> i == first(true_dims) ? n : 1, Val(ndims(x))),
     )
 
     project_x = ChainRulesCore.ProjectTo(x)
     function rfft_pullback(ȳ)
-        x̄ = project_x(brfft(ChainRulesCore.unthunk(ȳ) ./ scale, d, dims))
-        return ChainRulesCore.NoTangent(), x̄, ChainRulesCore.NoTangent()
+        x̄ = project_x(brfft(ChainRulesCore.unthunk(ȳ) ./ scale, d, dims_args...))
+        dims_args_tangent = (dims === nothing) ? () : (ChainRulesCore.NoTangent(),)
+        return ChainRulesCore.NoTangent(), x̄, dims_args_tangent... 
     end
     return y, rfft_pullback
 end
 
-function ChainRulesCore.frule((_, Δx, _), ::typeof(ifft), x::AbstractArray, dims)
-    y = ifft(x, dims)
-    Δy = ifft(Δx, dims)
+function ChainRulesCore.frule(Δargs, ::typeof(ifft), x::AbstractArray, dims=nothing)
+    Δx = Δargs[2]
+    args = (dims === nothing) ? () : (dims,)
+    y = ifft(x, args...)
+    Δy = ifft(Δx, args...)
     return y, Δy
 end
-function ChainRulesCore.rrule(::typeof(ifft), x::AbstractArray, dims)
-    y = ifft(x, dims)
-    invN = normalization(y, dims)
+function ChainRulesCore.rrule(::typeof(ifft), x::AbstractArray, dims=nothing)
+    dims_args = (dims === nothing) ? () : (dims,)
+    true_dims = (dims === nothing) ? (1:ndims(x)) : dims 
+    y = ifft(x, dims_args...)
+    invN = normalization(y, true_dims)
     project_x = ChainRulesCore.ProjectTo(x)
     function ifft_pullback(ȳ)
-        x̄ = project_x(invN .* fft(ChainRulesCore.unthunk(ȳ), dims))
-        return ChainRulesCore.NoTangent(), x̄, ChainRulesCore.NoTangent()
+        x̄ = project_x(invN .* fft(ChainRulesCore.unthunk(ȳ), dims_args...))
+        dims_args_tangent = (dims === nothing) ? () : (ChainRulesCore.NoTangent(),)
+        return ChainRulesCore.NoTangent(), x̄, dims_args_tangent...
     end
     return y, ifft_pullback
 end
 
-function ChainRulesCore.frule((_, Δx, _, _), ::typeof(irfft), x::AbstractArray, d::Int, dims)
-    y = irfft(x, d, dims)
-    Δy = irfft(Δx, d, dims)
+function ChainRulesCore.frule(Δargs, ::typeof(irfft), x::AbstractArray, d::Int, dims=nothing)
+    Δx = Δargs[2]
+    dims_args = (dims === nothing) ? () : (dims,)
+    y = irfft(x, d, dims_args...)
+    Δy = irfft(Δx, d, dims_args...)
     return y, Δy
 end
-function ChainRulesCore.rrule(::typeof(irfft), x::AbstractArray, d::Int, dims)
-    y = irfft(x, d, dims)
+function ChainRulesCore.rrule(::typeof(irfft), x::AbstractArray, d::Int, dims=nothing)
+    dims_args = (dims === nothing) ? () : (dims,)
+    true_dims = (dims === nothing) ? (1:ndims(x)) : dims 
```

Comment:
I could have avoided `dims_args` and just computed and passed on `true_dims` in all rules, but I decided to use the current approach to maintain the nice property that the call to the primal receives exactly the same keyword arguments as provided to the rule.